src/GraphQL/Utility/FileUpload.php

```diff
@@ -272,29 +280,32 @@ public function saveFileUpload(UploadedFile $uploaded_file, array $settings): Fi
       // before it is saved.
       $file->setSize(@filesize($temp_file_path));
 
-      // Validate against file_validate() first with the temporary path.
-      // @todo Drupal 10.1 compatibility, needs to be converted to file validate
-      // service later.
-      // @phpstan-ignore-next-line
-      $errors = file_validate($file, $validators);
+      // Validate against fileValidator first with the temporary path.
+      /** @var \Symfony\Component\Validator\ConstraintViolationListInterface $errors */
+      $errors = $this->fileValidator->validate($file, $validators);
+      if (count($errors) > 0) {
+        $response->addViolations($errors);
+        return $response;
+      }
+
+      // Validate Image resolution.
       $maxResolution = $settings['max_resolution'] ?? 0;
       $minResolution = $settings['min_resolution'] ?? 0;
       if (!empty($maxResolution) || !empty($minResolution)) {
-        $errors += $this->validateFileImageResolution($file, $maxResolution, $minResolution);
-      }
-
-      if (!empty($errors)) {
-        $response->addViolations($errors);
-        return $response;
+        $image_resolution_errors = $this->validateFileImageResolution($file, $maxResolution, $minResolution);
```

Comment:
@apathak18 We are returning response twice i.e first validating if path is valid and then next time we are validating image resolution. It would be great, if we can perform all validations and return all errors at once so that user is aware of what all errors they need to fix.

Reply:
Done, thanks!!